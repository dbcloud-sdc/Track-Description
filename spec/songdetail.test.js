
import React from 'react';
import Enzyme from 'enzyme';
import { mount, shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SongDetailWidget from '../client/src/components/songDetailWidget';

import artistData from '../client/src/data/artistData';
import songData from '../client/src/data/songData';

Enzyme.configure({ adapter: new Adapter() });


const toggleTruncate = () => false;
describe('SongDetailWidget', () => {
  it('should render correctly with dummy data', () => {
    const component = shallow(<SongDetailWidget artistData={artistData} songData={songData[0]} truncated toggleTruncate={toggleTruncate} />);
    expect(component).toMatchSnapshot();
  });

  it('should expand when clicking show more', () => {
    const component = shallow(<SongDetailWidget artistData={artistData} songData={songData[0]} truncated toggleTruncate={toggleTruncate} />);
    component.find('div.showMoreText')
      .simulate('click');
    expect(component.contains('Show more')).toEqual(true);
  });
});
