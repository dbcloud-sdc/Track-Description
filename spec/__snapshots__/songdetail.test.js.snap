// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongDetailWidget should render correctly with dummy data 1`] = `
<div
  className="songDetailContainer"
  id="songDetailContainer"
>
  <div
    className="songDetailWidgetSmall"
  >
    <div
      style={
        Object {
          "whiteSpace": "pre-wrap",
        }
      }
    >
      <div>
        Velit 
        quam 
        eos. 
        <span
          className="songDescriptionURL"
        >
          www.google.com 
           
        </span>
        <span
          className="songDescriptionAt tooltip"
        >
          @ipsum
           
          <span
            className="bottom"
          >
            <div
              className="popup"
            >
              <img
                src="https://s3.amazonaws.com/uifaces/faces/twitter/murrayswift/128.jpg"
              />
              <div
                className="songComponentAtArtistName"
                title="Visit ipsum's profile"
              >
                ipsum
              </div>
              <div
                className="songComponentAtFollows"
                title="2134608 followers"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  color="#999"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        640,
                        512,
                        Array [],
                        "f500",
                        "M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z",
                      ],
                      "iconName": "user-friends",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                 
                <small>
                  <NumericLabel
                    params={
                      Object {
                        "commafy": true,
                        "currency": false,
                        "justification": "L",
                        "locales": "en-AU",
                        "percentage": false,
                        "precision": 0,
                        "shortFormat": false,
                        "title": false,
                      }
                    }
                  >
                    2134608
                  </NumericLabel>
                </small>
              </div>
              <FollowButton
                isFollowing={0}
                toggleFollow=""
              />
            </div>
            <i />
          </span>
           
        </span>
        Mollitia 
        voluptatem 
        sunt 
        ut 
        sit 
        atque. 
        Et 
        qui 
        at 
        quas 
        in 
        sit 
        error 
        nihil. 
        Velit 
        illo 
        natus 
        ullam 
        et 
        dolor. 
        Quod 
        est 
        sit 
        delectus 
        laudantium. 
        Quisquam 
        magni 
        et 
        ut.
 
        
Omnis 
        voluptas 
        doloremque 
        harum 
        in 
        nemo. 
        Est 
        et 
        dolor 
        dolorem 
        voluptates 
        consequatur 
        voluptatem. 
        Sequi 
        qui 
        est 
        adipisci.
 
        
Sit 
        aut 
        voluptatem 
        iusto 
        ea 
        doloribus 
        est 
        soluta 
        soluta. 
        Et 
        nisi 
        esse 
        eveniet 
        ex 
        maiores 
        nemo. 
        Dolores 
        ex 
        ratione. 
        Qui 
        vero 
        vero 
        dolores 
        in 
        nulla. 
        ( 
        <span
          className="songDescriptionAt tooltip"
        >
          @harum
           
          <span
            className="bottom"
          >
            <div
              className="popup"
            >
              <img
                src="https://s3.amazonaws.com/uifaces/faces/twitter/robergd/128.jpg"
              />
              <div
                className="songComponentAtArtistName"
                title="Visit harum's profile"
              >
                harum
              </div>
              <div
                className="songComponentAtFollows"
                title="3500573 followers"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  color="#999"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        640,
                        512,
                        Array [],
                        "f500",
                        "M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z",
                      ],
                      "iconName": "user-friends",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                 
                <small>
                  <NumericLabel
                    params={
                      Object {
                        "commafy": true,
                        "currency": false,
                        "justification": "L",
                        "locales": "en-AU",
                        "percentage": false,
                        "precision": 0,
                        "shortFormat": false,
                        "title": false,
                      }
                    }
                  >
                    3500573
                  </NumericLabel>
                </small>
              </div>
              <FollowButton
                isFollowing={0}
                toggleFollow=""
              />
            </div>
            <i />
          </span>
           
        </span>
        )
 
        
Itaque 
        repellat 
        rerum 
        qui 
        iste 
        quam 
        praesentium 
        iusto. 
        Beatae 
        et 
        dolor 
        officia 
        vero 
        dolorem. 
        Assumenda 
        ea 
        temporibus 
        voluptatum 
        est 
        nemo 
        nihil 
        mollitia 
        alias 
        error. 
        Ut 
        est 
        quasi. 
        Sunt 
        repellendus 
        est 
        quia 
        quidem 
        placeat 
        beatae. 
        Nobis 
        voluptatem 
        molestias. 
      </div>
    </div>
    <div
      className="detailsContainer"
    >
      <SongDetail
        label="Released By:"
        value="ipsum"
      />
      <SongDetail
        label="Release Date:"
        value="2018-08-02T07:00:00.000Z"
      />
      <SongDetail
        label="P-line:"
        value="et inventore"
      />
      <SongDetail
        label="Licensed By:"
      />
    </div>
    <div />
    <div
      className="descriptionTagContainer"
    >
      <span
        className="tag"
      >
        #
        debitis
      </span>
      <span
        className="tag"
      >
        #
        possimus
      </span>
      <span
        className="tag"
      >
        #
        illo
      </span>
    </div>
  </div>
  <div
    className="showMoreText"
    onClick={[Function]}
  >
    Show more
    <span>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              320,
              512,
              Array [],
              "f107",
              "M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z",
            ],
            "iconName": "angle-down",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </span>
  </div>
</div>
`;
